src/chainstate/burn/operations/peg_in.rs

```diff
@@ -124,30 +131,24 @@ impl PegInOp {
         Ok(())
     }
 
-    /// Returns the leading non-zero bytes of the subslice `data[from..to]`
+    /// Returns the contract identifier and the number of bytes read
     ///
-    /// # Panics
-    ///
-    /// Panics if `from` is larger than or equal to `to`
-    fn leading_non_zero_bytes(data: &[u8], from: usize, to: usize) -> Option<&[u8]> {
-        assert!(from < to);
-
-        let end_of_non_zero_slice = {
-            let mut end = to.min(data.len());
-            for i in from..end {
-                if data[i] == 0 {
-                    end = i;
-                    break;
-                }
-            }
-            end
-        };
-
-        if from == end_of_non_zero_slice {
-            return None;
+    /// # Errors
+    /// - `Utf8Error` if the contract name is not valid UTF-8
+    /// - `ParseError` if the contract name is not a valid `ContractName`
+    fn read_contract_name(data: &[u8]) -> Result<Option<ContractName>, ParseError> {
+        let size = data[0] as usize;
```

Comment:
What's the point of having the size prefix if you're just going to inject `&data[1..41]` into a `ContractName`?  Shouldn't you instead be using `&data[1..(1 + size)]`, with a bounds-check on `size`?

Reply:
Sorry just saw this comment now. The size prefix makes it much easier to parse the data in Clarity, as per this discussion https://github.com/Trust-Machines/stacks-sbtc/pull/339#discussion_r1193026002